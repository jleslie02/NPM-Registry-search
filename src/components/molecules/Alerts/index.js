/* @jsx jsx */
import React from "react";
// eslint-disable-next-line no-unused-vars
import { jsx, css } from "@emotion/core";
import PropTypes from "prop-types";

const propTypes = {
  message: PropTypes.string,
  isError: PropTypes.bool
};

const defaultProps = {
  message: "",
  isError: false
};

const StatefulAlerts = props => {
  const { isError, message, theme } = props;
  const classes = {
    alert: css(
      (() => ({
        height: "70px",
        padding: "0 20px",
        background: "rgba(253, 116, 116, 0.57)",
        color: theme.palette.alert.color,
        ...theme.mixins.flexDisplay(),
        ...theme.mixins.alignItems("center")
      }))()
    )
  };
  if (!isError) return null;

  return (
    <div data-testid="alerts" css={classes.alert}>
      {message}
    </div>
  );
};

StatefulAlerts.propTypes = propTypes;
StatefulAlerts.defaultProps = defaultProps;

export default StatefulAlerts;
