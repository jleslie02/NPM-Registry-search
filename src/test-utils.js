import React from "react";
import { render } from "react-testing-library";
import makeAtomicTheme from "./theme";

export const themedRender = (MyComponent, props) => {
  return render(<MyComponent theme={makeAtomicTheme()} {...props} />);
};

export const themedComponent = (MyComponent, props) => {
  return <MyComponent theme={makeAtomicTheme()} {...props} />;
};
