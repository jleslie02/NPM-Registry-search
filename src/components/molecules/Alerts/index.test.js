// Static component with conditional rendering so snapshot test should suffice
import renderer from "react-test-renderer";
import Alerts from "./index";
import { themedComponent } from "../../../test-utils";

it("renders correctly with default props", () => {
  const props = {
    message: "",
    isError: false
  };
  const tree = renderer.create(themedComponent(Alerts, props)).toJSON();
  expect(tree).toBe(null);
});

it("renders nothing if there is no error to show", () => {
  const props = {
    message: "An error",
    isError: false
  };
  const tree = renderer.create(themedComponent(Alerts, props)).toJSON();
  expect(tree).toBe(null);
});

it("renders correctly with full props", () => {
  const props = {
    message: "An error",
    isError: true
  };
  const tree = renderer.create(themedComponent(Alerts, props)).toJSON();
  expect(tree).toMatchSnapshot();
});
