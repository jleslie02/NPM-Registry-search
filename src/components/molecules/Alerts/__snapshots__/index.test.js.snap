// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with full props 1`] = `
<div
  className="css-ufgy5p-StatefulAlerts"
  data-testid="alerts"
>
  An error
</div>
`;
